apps/dashboard/src/components/workflow-editor/edges.tsx

```diff
@@ -60,8 +63,18 @@ export function AddNodeEdge({
                     ];
 
                     update({
-                      ...workflow,
-                      steps: updatedSteps,
+                      data: {
+                        ...workflow,
+                        steps: updatedSteps,
+                      },
+                      onSuccess: (data) => {
+                        navigate(
+                          buildRoute(ROUTES.EDIT_STEP_TEMPLATE, {
+                            workflowSlug: workflow.slug,
+                            stepSlug: data.steps[indexToAdd].slug,
+                          })
```

Comment:
There is an issue here, when we add a non sidebar item like delay, it opens a random step email editor or custom controls

Reply:
Hmm ur right. I'll filter those out.